src/StreamrClient.js

```diff
@@ -277,8 +277,12 @@ export default class StreamrClient extends EventEmitter {
             return this._requestPublish(streamMessage, sessionToken)
         } else if (this.options.autoConnect) {
             this.publishQueue.push([streamId, data, timestamp, partitionKey])
-            return this.connect().catch(() => {}) // ignore
+            if (this.connection.state === Connection.State.DISCONNECTED) {
```

Comment:
This changes the logic if the connection happens to be `DISCONNECTING`. If `autoConnect` is true and we're publishing, we should probably (re-)connect in this case.

```suggestion
            if (this.connection.state === Connection.State.DISCONNECTED || this.connection.state === Connection.State.DISCONNECTING) {
```

Reply:
@hpihkala Currently it seems to hang if `connect()` is called while in `DISCONNECTING` state, see https://github.com/streamr-dev/streamr-client-javascript/pull/52

Reply:
Updated in 055f55a. Rather than testing for `DISCONNECTED` or `DISCONNECTING`, changed to `!== CONNECTING`, which is equivalent since we just tested for `CONNECTED` in the `isConnected()`.